app/src/main/java/ru/practicum/android/diploma/filters/presentation/ui/ChooseRegionFragment.kt

```diff
@@ -0,0 +1,181 @@
+package ru.practicum.android.diploma.filters.presentation.ui
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.setFragmentResult
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import ru.practicum.android.diploma.common.utils.debounce
+import ru.practicum.android.diploma.databinding.FragmentRegionsBinding
+import ru.practicum.android.diploma.filters.domain.models.ChooseRegionsResult
+import ru.practicum.android.diploma.filters.presentation.rv.RegionAdapter
+import ru.practicum.android.diploma.filters.presentation.view_model.FiltersViewModel
+import ru.practicum.android.diploma.search.data.dto.response_models.Area
+
+
+class ChooseRegionFragment : Fragment() {
+    private lateinit var binding: FragmentRegionsBinding
+    private lateinit var regionAdapter: RegionAdapter
+    private lateinit var textWatcher: TextWatcher
+    private val viewModel by viewModel<FiltersViewModel>()
+    private var regionList = ArrayList<Area>()
+    private var regionListSaved = ArrayList<Area>()
+    private var regionListFilter = ArrayList<Area>()
+    private lateinit var selected: Area
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        super.onCreateView(inflater, container, savedInstanceState)
+        binding = FragmentRegionsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initCountryAdapter()
+        getRegions()
+        observeViewModel()
+        initListeners()
+    }
+
+    private fun observeViewModel() {
+        viewModel.chooseRegionResult.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                ChooseRegionsResult.EmptyResult -> {
+                    showErrorList()
+                }
+
+                is ChooseRegionsResult.Error -> {
+                    showErrorList()
+                    println(state.exception)
```

Comment:
Лучше подключить Logger:
```suggestion
                    Log.d("ChooseRegionsResultError", state.exception)
```